src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -351,25 +347,15 @@ private static MetadataReference GetOrCreateMetadataReference(ref MetadataRefere
            Func<SyntaxNode, bool> syntaxNodePredicate = null,
            bool argumentOrderDoesNotMatter = false)
         {
-            return Diagnostic(
+            return TestHelpers.Diagnostic(
                 code,
-                NormalizeNewLines(squiggledText),
+                squiggledText,
                 arguments,
                 startLocation,
                 syntaxNodePredicate,
                 argumentOrderDoesNotMatter);
         }
 
-        public static string NormalizeNewLines(XCData data)
-        {
-            if (ExecutionConditionUtil.IsWindows)
-            {
-                return data.Value.Replace("\n", "\r\n");
-            }
-
-            return data.Value;
-        }
```

Comment:
oh wow.  boy do i hate this function.  I did not know this existed...

Reply:
sadly necessary given the strict implementation of the XML standard which requires \n new lines

---
In reply to: [217518865](https://github.com/dotnet/roslyn/pull/29852#discussion_r217518865) [](ancestors = 217518865)